src/api/server.js

```diff
@@ -0,0 +1,118 @@
+const express = require('express');
+const app = express();
+const MongoClient = require('mongodb').MongoClient
+var ObjectId = require('mongodb').ObjectID;
+const bodyParser = require('body-parser');
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+
+
+var mongoUrl = 'mongodb://localhost/test';
+var db
+
+MongoClient.connect(mongoUrl, (err, database) => {
+    if (err) return console.log(err)
+    db = database
+    app.listen(4000, () => {
+        console.log('listening on 4000')
+    })
+})
+
+// get all accounts
+app.get('/api/account', function(req, res) {
+    console.log("get all accounts");
+    var collection = db.collection('account');
+    collection.find().toArray(function(err, results) {
+        if (err) {
+            res.status(400).send("udapte error:" + JSON.stringify(err));
+        }
+        res.status(200).send(results);
+    });
+});
+
+// get account by id
+app.get('/api/account/:id', function(req, res) {
+    console.log("find by: " + req.params.id);
+    if (req.params.id.length != 12 && req.params.id.length != 24) {
+        res.status(404).send("account not found:");
+    } else {
+        var collection = db.collection('account');
+
+        collection.findOne({ "_id": new ObjectId(req.params.id) }, function(err, results) {
+            if (err) {
+                res.status(404).send("account not found:" + JSON.stringify(err));
+            } else if (results == null) {
+                res.status(404).send("account not found:");
+            } else {
+                res.status(200).send(results);
+            }
+        });
+    }
+});
+
+// remove account by id
+app.delete('/api/account/:id', function(req, res) {
+    console.log("remove by: " + req.params.id);
+    var collection = db.collection('account');
+    collection.remove({ "_id": new ObjectId(req.params.id) }, function(err, results) {
+        if (err) {
+            res.status(404).send("account not found:" + JSON.stringify(err));
+        } else if (results.length == 0) {
+            res.status(404).send("account not found:");
+        } else {
+            res.status(200).send(req.params.id + " was removed successfully");
+        }
+    });
+});
+
+// insert/update account
+app.post('/api/account', function(req, res) {
+    console.log("Insert or update");
+    var collection = db.collection('account');
+    if (req.body._id === undefined) {
```

Comment:
While this is technically correct, I would prefer the form `if (!req.body._id)` since that checks for all falsy values (specifically _both_ `null` and `undefined`).